### ISSUE TITLE
Cannot checkout: "Billing country is a required field" error

### ISSUE THREAD

**Issue by webworxcy** [woocommerce/woocommerce-gateway-paypal-express-checkout#724]
### Describe the bug
Paypal express check out error "Cannot checkout: "Billing country is a required field" on single item purchase in wordpress.

### To reproduce

1. select all variables for the item to purchse
1. click on the visa icon in order to pay by credit card without having to sign up with paypal
1. goes through to paypal, then opens the checkout page, click on place order, error appears order does not go through.

**Screenshots**

### Expected behavior

**Environment (please complete the following information):**
 - WordPress Version - 5.4.1
 - WooCommerce Version - 4.0.1 
 - PayPal (Express) Checkout Plugin Version 1.6.21
 - Browser [e.g. chrome, safari] and Version Opera
 - Any other plugins installed:
AddToAny Share Buttons
Akismet Anti-Spam
BackWPup
Facebook for WooCommerce
Hello Dolly
Jetpack by WordPress.com
Loginizer
Mailchimp for WooCommerce
Messenger Customer Chat
Really Simple SSL
WooCommerce PDF Invoices & Packing Slips
WooCommerce Services
WP Cerber Security, Antispam & Malware Scan

### Additional details

<details><summary>System status</summary>
  
  ```
  <!--If applicable, paste the system status here. Please ensure you redact or remove any identifying information. -->
  ```
</details>

**Comment by mattallan** [comment]
Hi @webworxcy thanks for the report!

> check out error Cannot checkout: "Billing country is a required field

I'm unable to replicate this issue on my own site.

To help us find the cause:

1. Is there a specific country you are testing with which causes this issue?

2. Do you have a staging site available? If so, can you try rule out if this is caused by a plugin/theme conflict by only activating WooCommerce and PayPal Checkout and switching to a default theme?

**Comment by james-allan** [comment]
@webworxcy on top of what @mattallan asked above, could you let us know what type of products you sell (physical, virtual, downloadable), and whether you have the require billing address setting enabled. 

To find the billing address setting follow these steps:

1. Go to **WooCommerce Settings > Payments > PayPal Checkout** on your Wordpress admin dashboard.
2. Scroll down to Advanced settings (example below).

<img width="1560" alt="Screen Shot 2020-05-05 at 11 23 23 am" src="https://user-images.githubusercontent.com/8490476/81028026-33614100-8ec3-11ea-8b45-c5727fe9f010.png">

**Comment by webworxcy** [comment]
Hi Guys thank you for the reply, 

@mattallan 
Is there a specific country you are testing with which causes this issue? I have only tried Cyprus which is local to the store.

Do you have a staging site available? No but I did deactivate all other plugins and installed and used storefront theme as I had read in another forum, and still got the same error. 

@james-allan 
We are selling physical products and "required billing address" is not enabled.

Thank you

**Comment by mattallan** [comment]
Thanks @webworxcy for the extra information.

Given this I'm still not able to replicate the "Billing country is a required field" issue you're seeing but I am however getting a different error when I try to test using Cyprus.

When I place the order I get:

> Billing State / County is a required field.

Can you confirm if you meant County? and not Country? If so I think this is a similar issue to #722 

What's happening is, when you choose Cyprus in WooCommerce, the checkout loads both a `billing_state` field and a `billing_city` field, and both are required: https://d.pr/i/FGmyzM.

When you choose Cyprus in PayPal you only need to fill in the state field: https://d.pr/i/B0n9wX

Therefore when we return from PayPal that required city field does not exist and we throw an error.

One option we have for fixing this is to manually load the billing state field (and any field) if it's required and we don't get it from PayPal.

---

While we work on a fix, one option you have is to filter the `billing_state` field to make it optional ([following this guide](https://docs.woocommerce.com/document/tutorial-customising-checkout-fields-using-actions-and-filters/))

**Comment by webworxcy** [comment]
Hi @mattallan sorry you are right its is county not country sorry for that! 
I have created a staging site at https://myserendipity.store/1588685384692

**Comment by mattallan** [comment]
Thanks for confirming this @webworxcy. We have fixed this in #725 which will go out in our next release. Closing this
